.github/workflows/test.yml

```diff
@@ -0,0 +1,355 @@
+    outputs:
+      run-all: ${{
+          endsWith(github.ref, '/master') ||
+          endsWith(github.ref, '/develop') ||
+          contains(github.ref, '/release/')
+        }}
+      skip-all: ${{
+          (github.event_name == 'pull_request' || steps.has-pr.outputs.result == 'true') &&
+          github.event_name == 'push'
+        }}
+      build-handsontable-es-cjs: ${{ steps.path-filter.outputs.build-handsontable-es-cjs }}
+      build-handsontable-umd: ${{ steps.path-filter.outputs.build-handsontable-umd }}
+      test-handsontable-unit: ${{ steps.path-filter.outputs.test-handsontable-unit }}
+      test-handsontable-types: ${{ steps.path-filter.outputs.test-handsontable-types }}
+      test-handsontable-walkontable: ${{ steps.path-filter.outputs.test-handsontable-walkontable }}
+      test-handsontable-e2e: ${{ steps.path-filter.outputs.test-handsontable-e2e }}
+      test-handsontable-e2e-min: ${{ steps.path-filter.outputs.test-handsontable-e2e-min }}
+      test-angular: ${{ steps.path-filter.outputs.test-angular }}
+      test-react: ${{ steps.path-filter.outputs.test-react }}
+      test-vue: ${{ steps.path-filter.outputs.test-vue }}
+    steps:
+      - uses: actions/github-script@a3e7071a34d7e1f219a8a4de9a5e0a34d1ee1293 # https://github.com/actions/github-script/releases/tag/v4.0.2
+        if: ${{ github.event_name != 'pull_request' }}
+        id: has-pr
+        with:
+          script: |
+            const pullR = await github.pulls.list({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              head: `${context.repo.owner}:${{ github.ref }}`
+            });
+
+            return !!pullR.data.length;
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # https://github.com/actions/checkout/releases/tag/v2.3.4
+        if: ${{ github.event_name == 'push' }}
+      - uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721 # https://github.com/dorny/paths-filter/releases/tag/v2.10.2
+        if: ${{ !(steps.has-pr.outputs.result == 'true' && github.event_name == 'push') }}
+        id: path-filter
+        with:
+          filters: |
+            build-handsontable-umd:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'test/e2e/**'
+              - 'test/scripts/**'
+            build-handsontable-es-cjs:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'wrappers/**'
+            test-handsontable-unit:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'test/unit/**'
+            test-handsontable-types:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'test/types/**'
+            test-handsontable-walkontable:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'test/scripts/**'
```

Comment:
```suggestion
              - '.config/**'
              - 'src/3rdparty/walkontable/**'
```

Reply:
Are you sure we can remove these two? `test/scripts/**` can potentially change the way `puppeteer` runs tests, right?

Reply:
Right, the `test/scripts/**` entry can be helpful here.